### PR TITLE
[docs] Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by anik120**
Include instructions for setting up a dev environment,
conventions on commit message formatting, and other 
resources to help get contributions into marketplace-operator.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# How to contribute
+
+Operator Marketplace is Apache 2.0 licensed and accepts contributions via GitHub pull requests. This document outlines some of the conventions on commit message formatting, instructions on how to set up a dev environment, and other resources to help get contributions into operator-marketplace.
+
+# Email and Chat
+
+- Email: [operator-framework](https://groups.google.com/forum/#!forum/operator-framework)  
+
+## Setting up a dev environment
```
Comment (kevinrizza) [inline CONTRIBUTING.md]: Marketplace also works on vanilla Kubernetes. Should we at least mention that here? Just that, this is an example environment and vanilla kube and `kubectl` should work in addition to `oc` ?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+We follow a rough convention for commit messages that is designed to answer two
+questions: what changed and why. The subject line should feature the what and
+the body of the commit should describe the why.
+
+```
+scripts: add the test-cluster command
+
+this uses tmux to setup a test cluster that can easily be killed and started for debugging.
+
+Fixes #38
```
Comment (kevinrizza) [inline CONTRIBUTING.md]: This section seems irrelevant to `operator-marketplace`. What is # 38 and why are we referencing tmux? and a test-cluster command?

Comment (anik120) [reply CONTRIBUTING.md]: @kevinrizza this is suppose to be an example commit message. I've updated this section with a commit message from our repo to make it more relevant.

File: CONTRIBUTING.md (hunk 1 above)
Comment (anik120) [reply CONTRIBUTING.md]: @kevinrizza do we have a doc detailing steps that can be taken to get marketplace-operator running on vanilla Kubernetes (installing olm -> marketplace-operator)? I was thinking about writing something along the lines of:
```
This section provides detailed instructions on how to set up a development environment on an Openshift cluster. To use maketplace-operator in a vanilla Kubernetes cluster instead, follow the instructions [here] (<link-to-k8s-installation-doc>)  and use the `kubectl`/`oc` command. 
```
But I couldn't find any in the docs folder.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+Operator Marketplace is Apache 2.0 licensed and accepts contributions via GitHub pull requests. This document outlines some of the conventions on commit message formatting, instructions on how to set up a dev environment, and other resources to help get contributions into operator-marketplace.
+
+# Email and Chat
+
+- Email: [operator-framework](https://groups.google.com/forum/#!forum/operator-framework)  
+
+## Setting up a dev environment
+
+- Fork the repository on GitHub
+- Clone the repository in your go path.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: `forked repository`

File: CONTRIBUTING.md (hunk 1 above)
Comment (aravindhp) [reply CONTRIBUTING.md]: >But I couldn't find any in the docs folder.

Isn't that what you are writing :-)

This definitely needs an OpenShift vs plain vanilla Kuberenetes sections to call out the differences.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+- Get access to an Openshift cluster. The marketplace-operator is currently supported on Openshift 4.0. You can find instructions on installing an Openshift cluster [here](https://github.com/openshift/installer).
+- Turn off the Cluster-Service-Operator (CVO). The CVO manages the lifecycle of the marketplace-operator along with other internal Openshift components. Once the CVO is truned off, the marketplace-operator deployment can be deleted.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Please don't turn off CVO. Instead provide a `ClusterVersion` with `overrides`.
```
apiVersion: config.openshift.io/v1
kind: ClusterVersion
metadata:
  name: version
spec:
  channel: fast
  upstream: https://api.openshift.com/api/upgrades_info/v1/graph
  overrides:
    - kind: Deployment
      name: marketplace-operator
      namespace: openshift-marketplace
      unmanaged: true
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+- Fork the repository on GitHub
+- Clone the repository in your go path.
+- Get access to an Openshift cluster. The marketplace-operator is currently supported on Openshift 4.0. You can find instructions on installing an Openshift cluster [here](https://github.com/openshift/installer).
+- Turn off the Cluster-Service-Operator (CVO). The CVO manages the lifecycle of the marketplace-operator along with other internal Openshift components. Once the CVO is truned off, the marketplace-operator deployment can be deleted. 
+    `oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator`
+- Delete the marketplace-operator deployment `oc delete deployment openshift-marketplace -n marketplace-operator`
+- Install [Operator-SDK](https://github.com/operator-framework/operator-sdk).
+- `cd` into the marketplace-operator repository and `dep ensure` to download all dependencies.
+- `operator-sdk up local --namespace=openshift-marketplace --kubeconfig=<path-to-kubeconfig-file>` to complie the marketplace-operator and start the operator in you dev environment.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Doing all the testing in this manner will not test RBAC changes if that is being made. For that you also need instructions on how to use your own `Deployment` where you have replaced the image with the one you are building using the SDK.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+## Setting up a dev environment
+
+- Fork the repository on GitHub
+- Clone the repository in your go path.
+- Get access to an Openshift cluster. The marketplace-operator is currently supported on Openshift 4.0. You can find instructions on installing an Openshift cluster [here](https://github.com/openshift/installer).
+- Turn off the Cluster-Service-Operator (CVO). The CVO manages the lifecycle of the marketplace-operator along with other internal Openshift components. Once the CVO is truned off, the marketplace-operator deployment can be deleted. 
+    `oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator`
+- Delete the marketplace-operator deployment `oc delete deployment openshift-marketplace -n marketplace-operator`
+- Install [Operator-SDK](https://github.com/operator-framework/operator-sdk).
+- `cd` into the marketplace-operator repository and `dep ensure` to download all dependencies.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Why is this needed? We should have everything in the `vendor` folder.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+- `operator-sdk up local --namespace=openshift-marketplace --kubeconfig=<path-to-kubeconfig-file>` to complie the marketplace-operator and start the operator in you dev environment.      
+
+## Reporting bugs and creating issues
+
+Reporting bugs is one of the best ways to contribute. If any part of the operator-marketplace project has bugs or documentation mistakes, please let us know by [opening an issue](https://github.com/operator-framework/operator-marketplace/issues/new)
```
Comment (aravindhp) [inline CONTRIBUTING.md]: I suggest leaving out `Reporting bugs is one of the best ways to contribute.`

`opening an issue or a PR`

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+## Contribution flow
+
+This is a rough outline of what a contributor's workflow looks like:
+
+- Create a topic branch from where to base the contribution. This is usually master.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Mention that if you want to sync up with the upstream master, that a rebase workflow must be followed.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
+# How to contribute
+
+Operator Marketplace is Apache 2.0 licensed and accepts contributions via GitHub pull requests. This document outlines some of the conventions on commit message formatting, instructions on how to set up a dev environment, and other resources to help get contributions into operator-marketplace.
+
+# Email and Chat
```
Comment (kevinrizza) [inline CONTRIBUTING.md]: There's no email listed here under the Email header. Additionally, that link is to the operator-framework google group which we don't control and afaik even look at. We don't have a public email list set up either, so is this header necessary? Maybe we should set up something and add it later?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+```
+<subsystem>: <what changed>
+<BLANK LINE>
+<why this change was made>
+<BLANK LINE>
+<footer>
+```
+
+The first line is the subject and should be no longer than 50 characters, the second line is always blank, and other lines should be wrapped at 72 characters. This allows the message to be easier to read on GitHub as well as in various git tools.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Are you following the 72 character limit in your commit message? :-)

File: CONTRIBUTING.md (hunk 1 above)
Comment (kevinrizza) [reply CONTRIBUTING.md]: We don't have a doc, but the process should be identical except using `kubectl` instead of `oc`. The operator-sdk works with either.

File: CONTRIBUTING.md (hunk 9 above)
Comment (aravindhp) [reply CONTRIBUTING.md]: That is a good point. Main way to reach out should be issues I guess.

File: CONTRIBUTING.md (hunk 5 above)
Comment (kevinrizza) [reply CONTRIBUTING.md]: +1

File: CONTRIBUTING.md (hunk 6 above)
Comment (kevinrizza) [reply CONTRIBUTING.md]: @aravindhp to build yes, but for whatever reason the sdk does not read those dependencies locally. at least that is what we found on friday when trying to get this to run in Anik's local environment.

Comment (aravindhp) [reply CONTRIBUTING.md]: That should be called out here. This should also be brought up with the SDK team.

Comment (anik120) [reply CONTRIBUTING.md]: @kevinrizza I think that might have been due to the fact that my cloned repo was not in my go path.

File: CONTRIBUTING.md (hunk 10 above)
Comment (anik120) [reply CONTRIBUTING.md]: @aravindhp For the other PRs yes. For this one I edited my message now and it's following the convention :)

File: CONTRIBUTING.md (hunk 6 above)
Comment (kevinrizza) [reply CONTRIBUTING.md]: @anik120 did you confirm with a clean environment?

**Comment by anik120** [comment]
@kevinrizza @aravindhp addressed your comments. Discussed testing changes to RBAC with Kevin and penned down a few lines about that. Let me know if those lines are missing anything/not correct.

File: examples/cvo.override.yaml [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: version
+spec:
+  channel: fast
+  upstream: https://api.openshift.com/api/upgrades_info/v1/graph
+  overrides:
+    - kind: Deployment
+      name: marketplace-operator
+      namespace: openshift-marketplace
+      unmanaged: true
```
Comment (kevinrizza) [inline examples/cvo.override.yaml]: Missing newline at the end of the file.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+- Install [Operator-SDK](https://github.com/operator-framework/operator-sdk).
+- `operator-sdk up local --namespace=openshift-marketplace --kubeconfig=<path-to-kubeconfig-file>` to complie the marketplace-operator and start the operator in your dev environment. 
+
+### Testing changes to RBAC policy
+- If there's any modification in existing RBAC poilicies, the modifications need to be tested using a seperate deployment. Easiest way to do this is to do a local `docker build` and then push the image to a registry. Replace the [image here](https://github.com/operator-framework/operator-marketplace/blob/master/deploy/operator.yaml#L23) with the newly built image, then do a `oc apply -f deploy/` to reinstall the operator.  
+
+## Marketplace-operator on vanilla Kubernetes
+
+- To set up marketplace-operator on vanilla Kubernetes, [install Operator-Lifecycle-Manager(OLM)](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/install/install.md) in a Kubernetes Cluster. 
+- `cd` into the makertplace-operator repository and install the marketplace-operator using `kubectl apply -f deploy/upstream`.
```
Comment (kevinrizza) [inline CONTRIBUTING.md]: should be marketplace-operator

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+- The Cluster-Version-Operator(CVO) manages the lifecycle of the marketplace-operator along with other internal Openshift components. To delete the marketplace-operator deployment, insruct CVO to stop managing the marketplace-operator by applying the configuration file that can be found [here](examples/cvo.override.yaml). 
+- Delete the marketplace-operator deployment `oc delete deployment openshift-marketplace -n marketplace-operator`.
+- Install [Operator-SDK](https://github.com/operator-framework/operator-sdk).
+- `operator-sdk up local --namespace=openshift-marketplace --kubeconfig=<path-to-kubeconfig-file>` to complie the marketplace-operator and start the operator in your dev environment. 
+
+### Testing changes to RBAC policy
+- If there's any modification in existing RBAC poilicies, the modifications need to be tested using a seperate deployment. Easiest way to do this is to do a local `docker build` and then push the image to a registry. Replace the [image here](https://github.com/operator-framework/operator-marketplace/blob/master/deploy/operator.yaml#L23) with the newly built image, then do a `oc apply -f deploy/` to reinstall the operator.
```
Comment (kevinrizza) [inline CONTRIBUTING.md]: `separate`

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+- The Cluster-Version-Operator(CVO) manages the lifecycle of the marketplace-operator along with other internal Openshift components. To delete the marketplace-operator deployment, insruct CVO to stop managing the marketplace-operator by applying the configuration file that can be found [here](examples/cvo.override.yaml). 
+- Delete the marketplace-operator deployment `oc delete deployment openshift-marketplace -n marketplace-operator`.
+- Install [Operator-SDK](https://github.com/operator-framework/operator-sdk).
+- `operator-sdk up local --namespace=openshift-marketplace --kubeconfig=<path-to-kubeconfig-file>` to complie the marketplace-operator and start the operator in your dev environment. 
+
+### Testing changes to RBAC policy
+- If there's any modification in existing RBAC poilicies, the modifications need to be tested using a seperate deployment. Easiest way to do this is to do a local `docker build` and then push the image to a registry. Replace the [image here](https://github.com/operator-framework/operator-marketplace/blob/master/deploy/operator.yaml#L23) with the newly built image, then do a `oc apply -f deploy/` to reinstall the operator.  
+
```
Comment (kevinrizza) [inline CONTRIBUTING.md]: Can you specifically call out to use the `Dockerfile` in the root of the repo here? There are several dockerfiles in the project so it could be confusing.

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
+# How to contribute
+
+Operator Marketplace is Apache 2.0 licensed and accepts contributions via GitHub pull requests. This document outlines some of the conventions on commit message formatting, instructions on how to set up a dev environment, and other resources to help get contributions into operator-marketplace.  
+
+## Setting up a dev environment
+
+- Fork the repository on GitHub
+- Clone the forked repository in your go path.
+- Get access to an Openshift cluster. The marketplace-operator is currently supported on Openshift 4.0. You can find instructions on installing an Openshift cluster [here](https://github.com/openshift/installer).
+- The Cluster-Version-Operator(CVO) manages the lifecycle of the marketplace-operator along with other internal Openshift components. To delete the marketplace-operator deployment, insruct CVO to stop managing the marketplace-operator by applying the configuration file that can be found [here](examples/cvo.override.yaml).
```
Comment (kevinrizza) [inline CONTRIBUTING.md]: `To delete the marketplace-operator deployment` makes it sound like adding the override deletes the deployment. Can it be something like `Before deleting the marketplace-operator deployment, instruct CVO to ....(etc)` ?

File: CONTRIBUTING.md (hunk 13 above)
Comment (aravindhp) [inline CONTRIBUTING.md]: `oc apply -f deploy/operator.yaml`

File: CONTRIBUTING.md (hunk 15 above)
Comment (aravindhp) [inline CONTRIBUTING.md]: Please add the command to apply the override given you are doing this everywhere else.

File: CONTRIBUTING.md (hunk 13 above)
Comment (aravindhp) [inline CONTRIBUTING.md]: Why `docker build` vs `operator sdk build`?

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+
+## Marketplace-operator on vanilla Kubernetes
+
+- To set up marketplace-operator on vanilla Kubernetes, [install Operator-Lifecycle-Manager(OLM)](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/install/install.md) in a Kubernetes Cluster. 
+- `cd` into the makertplace-operator repository and install the marketplace-operator using `kubectl apply -f deploy/upstream`.
+- To delete the marketplace-operator deployment use `oc delete deployment marketplace`. 
+
+### Testing changes to RBAC policy
+- If there's any modification in existing RBAC poilicies, the modifications need to be tested using a seperate deployment. Easiest way to do this is to do a local `docker build` and then push the image to a registry. Replace the [image here](https://github.com/operator-framework/operator-marketplace/blob/master/deploy/upstream/07_operator.yaml#L28) with the newly built image, then do a `kubectl apply -f deploy/upstream/`.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Same comment as above.

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+
+## Reporting bugs and creating issues
+
+If any part of the operator-marketplace project has bugs or documentation mistakes, please let us know by [opening an issue](https://github.com/operator-framework/operator-marketplace/issues/new) or a PR. 
+
+## Contribution flow
+
+This is a rough outline of what a contributor's workflow looks like:
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Please remove rough

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+- Push changes in a topic branch to a personal fork of the repository.
+- To make sure that your topic branch is in sync with the remote master branch, follow a rebase workflow.
+- Submit a pull request to operator-framework/operator-marketplace.
+- The PR must receive a LGTM from two maintainers of the operator-marketplace project.
+
+Thanks for contributing!
+
+### Format of the commit message
+
+We follow a rough convention for commit messages that is designed to answer two
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Please remove rough

File: CONTRIBUTING.md (hunk 10 above)
Comment (aravindhp) [reply CONTRIBUTING.md]: We can go with 80 characters instead of 72.

File: examples/cvo.override.yaml (hunk 11 above)
Comment (aravindhp) [inline examples/cvo.override.yaml]: Please add a newline.

**Comment by anik120** [comment]
@kevinrizza @aravindhp addressed all comments.

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+- Delete the marketplace-operator deployment with `oc delete deployment openshift-marketplace -n marketplace-operator`.
+- Install [Operator-SDK](https://github.com/operator-framework/operator-sdk).
+- `operator-sdk up local --namespace=openshift-marketplace --kubeconfig=<path-to-kubeconfig-file>` to complie the marketplace-operator and start the operator in your dev environment. 
+
+### Testing changes to RBAC policy
+- If there's any modification in existing RBAC poilicies, the modifications need to be tested using a separate deployment. Easiest way to do this is to do a local `operator-sdk build` using [this dockerfile](./Dockerfile) and then `docker push` the image to a registry. Replace the [image here](https://github.com/operator-framework/operator-marketplace/blob/master/deploy/operator.yaml#L23) with the newly built image, then do a `oc apply -f deploy/operator.yaml` to reinstall the operator.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Please add a section using ``` listing the workflow. I find it hard to follow by reading all the text. Moreover doesn't `operator-sdk build` pick the `Dockerfile`? So you just need to mention that it ends up using that particular `Dockerfile`.

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+## Reporting bugs and creating issues
+
+If any part of the operator-marketplace project has bugs or documentation mistakes, please let us know by [opening an issue](https://github.com/operator-framework/operator-marketplace/issues/new) or a PR. 
+
+## Contribution flow
+
+This is an outline of what a contributor's workflow looks like:
+
+- Create a topic branch from where to base the contribution. This is usually master.
+- Make commits of logical units.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: This needs to be expanded. Please mention that this means a commit should typically try to do only one thing. For example fixing a bug and adding a feature shouldn't be in the same commit. Mixing bugs and feature shouldn't be done in the same PR for that. A person's individual workflow adding a change shouldn't be reflected in the commits.

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+- Make sure commit messages are in the proper format (see below).
+- Push changes in a topic branch to a personal fork of the repository.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Shouldn't the first step be to for the repo?

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -0,0 +1,70 @@
...
+- To make sure that your topic branch is in sync with the remote master branch, follow a rebase workflow.
+- Submit a pull request to operator-framework/operator-marketplace.
+- The PR must receive a LGTM from two maintainers of the operator-marketplace project.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: It is one /lgtm and one /approve from the maintainers. We try to give two /lgtm but there is nothing in the CI process that looks for that.

File: CONTRIBUTING.md (hunk 19 above)
Comment (anik120) [reply CONTRIBUTING.md]: @aravindhp restructured the document. Let me know what you think now.

File: CONTRIBUTING.md (hunk 21 above)
Comment (anik120) [reply CONTRIBUTING.md]: @aravindhp could you repeat this.

Comment (aravindhp) [reply CONTRIBUTING.md]: Sorry, shouldn't forking the repo be the first step?

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -0,0 +1,117 @@
...
+- The Cluster-Version-Operator(CVO) manages the lifecycle of the marketplace-operator along with other internal Openshift components. Before deleting the marketplace-operator deployment, insruct CVO to stop managing the marketplace-operator 
+```
+$ oc apply -f examples/cvo.override.yaml
+``` 
+- Delete the marketplace-operator deployment
+```
+$ oc delete deployment openshift-marketplace -n marketplace-operator
+```
+- Install [Operator-SDK](https://github.com/operator-framework/operator-sdk).
+- Complie the marketplace-operator and start the operator in your dev environment
```
Comment (aravindhp) [inline CONTRIBUTING.md]: `Compile`

File: CONTRIBUTING.md [hunk 24]
```diff
@@ -0,0 +1,117 @@
+# How to contribute
+
+Operator Marketplace is Apache 2.0 licensed and accepts contributions via GitHub pull requests. This document outlines some of the conventions on commit message formatting, instructions on how to set up a dev environment, and other resources to help get contributions into operator-marketplace.  
+
+## Setting up a dev environment
+
+- Fork the repository on GitHub
+- Clone the forked repository in your go path.
+- Get access to an Openshift cluster. The marketplace-operator is currently supported on Openshift 4.0. You can find instructions on installing an Openshift cluster [here](https://github.com/openshift/installer).
+- The Cluster-Version-Operator(CVO) manages the lifecycle of the marketplace-operator along with other internal Openshift components. Before deleting the marketplace-operator deployment, insruct CVO to stop managing the marketplace-operator
```
Comment (aravindhp) [inline CONTRIBUTING.md]: `instruct CVO`

File: CONTRIBUTING.md [hunk 25]
```diff
@@ -0,0 +1,117 @@
...
+$ oc delete deployment openshift-marketplace -n marketplace-operator
+```
+- Install [Operator-SDK](https://github.com/operator-framework/operator-sdk).
+- Complie the marketplace-operator and start the operator in your dev environment
+```
+$ operator-sdk up local --namespace=openshift-marketplace --kubeconfig=<path-to-kubeconfig-file>  
+```
+
+### Testing changes to RBAC policy
+- If there's any modification in existing RBAC poilicies, the modifications need to be tested using a separate deployment. Easiest way to do this is to build an image locally using [this dockerfile](./Dockerfile), push the image to a registry, and replace the [image here](https://github.com/operator-framework/operator-marketplace/blob/master/deploy/operator.yaml#L23) with the newly built image. The operator can then be redeployed with the new changes.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: `policies`

File: CONTRIBUTING.md [hunk 26]
```diff
@@ -0,0 +1,117 @@
...
+```
+$ kubectl apply -f deploy/upstream
+```
+- To delete the marketplace-operator deployment use 
+```
+$ oc delete deployment marketplace
+``` 
+
+### Testing changes to RBAC policy
+- If there's any modification in existing RBAC poilicies, the modifications need to be tested using a seperate deployment. Easiest way to do this is to build an image locally using [this dockerfile](./Dockerfile), push the image to a registry, and replace the [image here](https://github.com/operator-framework/operator-marketplace/blob/master/deploy/upstream/07_operator.yaml#L28) with the newly built image. The operator can then be redeployed with the new changes.
```
Comment (aravindhp) [inline CONTRIBUTING.md]: `policies`
`separate`

File: CONTRIBUTING.md [hunk 27]
```diff
@@ -0,0 +1,117 @@
...
+
+```
+$ cd <path-to-operator-marketplace-repo>
+
+$ operator-sdk build . --tag=<REGISTRYHOST>/<USERNAME>/marketplace-operator
+
+$ docker push <REGISTRYHOST>/<USERNAME>/marketplace-operator 
+
+$ vi deploy/upstream/07_operator.yaml
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Please use a `sed` here to do the replacement

**Comment by anik120** [comment]
@aravindhp I think it's safe to say all the spell-checking tools out there ruined my typing skills. Corrected all the errors. Also added instructions to use sed to replace images.

File: CONTRIBUTING.md (hunk 21 above)
Comment (anik120) [reply CONTRIBUTING.md]: @aravindhp I think I have listed forking the repo as the first step. Please let me know if you can see otherwise.

File: CONTRIBUTING.md [hunk 28]
```diff
@@ -0,0 +1,109 @@
...
+- If there's any modification in existing RBAC policies, the modifications need to be tested using a separate deployment. Easiest way to do this is to build an image locally using [this dockerfile](./Dockerfile), push the image to a registry, and replace the [image here](https://github.com/operator-framework/operator-marketplace/blob/master/deploy/operator.yaml#L23) with the newly built image. The operator can then be redeployed with the new changes.   
+
+```
+$ cd <path-to-operator-marketplace-repo>
+
+$ operator-sdk build . --tag=<REGISTRYHOST>/<USERNAME>/marketplace-operator
+
+$ docker push <REGISTRYHOST>/<USERNAME>/marketplace-operator 
+
+$ sed -i -e 's?quay.io/openshift/origin-operator-marketplace:latest?quay.io/anik120/marketplace-operator?g' deploy/operator.yaml
```
Comment (aravindhp) [inline CONTRIBUTING.md]: Please use `<REGISTRYHOST>/<USERNAME>/` like you have in other places.

File: CONTRIBUTING.md (hunk 21 above)
Comment (aravindhp) [reply CONTRIBUTING.md]: You have in `Setting up a dev environment`. Here you just need to mention that you are creating a topic branch in the forked repo in your first bullet.

Comment (anik120) [reply CONTRIBUTING.md]: @aravindhp Ah right. Added that now.

File: CONTRIBUTING.md [hunk 29]
```diff
@@ -0,0 +1,111 @@
+# How to contribute
+
+Operator Marketplace is Apache 2.0 licensed and accepts contributions via GitHub pull requests. This document outlines some of the conventions on commit message formatting, instructions on how to set up a dev environment, and other resources to help get contributions into operator-marketplace.  
+
+## Setting up a dev environment
+
+- Fork the repository on GitHub
+- Clone the forked repository in your go path.
+- Get access to an Openshift cluster. The marketplace-operator is currently supported on Openshift 4.0. You can find instructions on installing an Openshift cluster [here](https://github.com/openshift/installer).
```
Comment (SamiSousa) [inline CONTRIBUTING.md]: nit: Should it be `OpenShift` or `Openshift`?

File: CONTRIBUTING.md [hunk 30]
```diff
@@ -0,0 +1,111 @@
...
+- `cd` into the marketplace-operator repository and install the marketplace-operator using 
+```
+$ kubectl apply -f deploy/upstream
+```
+- To delete the marketplace-operator deployment use 
+```
+$ oc delete deployment marketplace
+``` 
+
+### Testing changes to RBAC policy
```
Comment (SamiSousa) [inline CONTRIBUTING.md]: nit: Since this section is specific for testing RBAC on vanilla kube, can the header reflect this? I was scrolling through and thought this was the same as the section for OpenShift, but this is specific for kubernetes. On a quick scroll it's tough to distinguish the two sections.

File: CONTRIBUTING.md [hunk 31]
```diff
@@ -0,0 +1,111 @@
...
+```
+$ cd <path-to-operator-marketplace-repo>
+
+$ operator-sdk build . --tag=<REGISTRYHOST>/<USERNAME>/marketplace-operator
+
+$ docker push <REGISTRYHOST>/<USERNAME>/marketplace-operator 
+
+$ sed -i -e 's?quay.io/openshift/origin-operator-marketplace:latest?<REGISTRYHOST>/<USERNAME>/marketplace-operator?g' deploy/upstream/07_operator.yaml
+
+$ oc apply -f deploy/upstream/operator.yaml
```
Comment (SamiSousa) [inline CONTRIBUTING.md]: `kubectl apply`

Comment (anik120) [reply CONTRIBUTING.md]: @SamiSousa thank you for noticing this. Addressed it.

File: CONTRIBUTING.md (hunk 30 above)
Comment (anik120) [reply CONTRIBUTING.md]: @SamiSousa changed the title. Please let me know if it's easier to distinguish now.

File: CONTRIBUTING.md (hunk 29 above)
Comment (anik120) [reply CONTRIBUTING.md]: @SamiSousa looks like it's OpenShift :)
